src/pages/MainPage.vue

```diff
@@ -422,14 +427,14 @@ export default {
       }
     },
     setActiveRoad: function () {
+      const roadRequested = this.$route.params.road;
       if (this.roads.hasOwnProperty(this.$route.params.road)) {
-        const roadRequested = this.$route.params.road;
-        if (roadRequested in this.roads) {
-          this.$store.commit('setActiveRoad', roadRequested);
-          return true;
-        }
+        this.$store.commit('setActiveRoad', roadRequested);
+        return true;
+      } else if (!this.$cookies.isKey('accessInfo')) {
+        // If user isn't logged in, and bad road id in url, then redirect to default road
+        this.$router.push({ path: '/' });
```

Comment:
I'm not sure if it's this line that's causing it, but there is a new change in behavior where navigating to either '/road' or '/road/[first road id]' both do the same thing.  Previously '/road' got expanded to '/road/[first road id]' so there was always a consistent URL.  Can you look into what caused this change?

Reply:
What's a good way to know what is the first road's id so I can redirect to it? Seems like in the store, they're stored in an object not an array.

Reply:
I believe that Object.keys() and iteration always produce the same ordering in javascript, so you should just be able to take the first entry in Object.keys() of the roads.